src/Bicep.LangServer/Completions/BicepCompletionContext.cs

```diff
@@ -212,13 +212,28 @@ public static BicepCompletionContext Create(IFeatureProvider featureProvider, Co
                 // check if we're inside an expression
                 kind |= ConvertFlag(IsInnerExpressionContext(matchingNodes, offset), BicepCompletionContextKind.Expression);
 
-                if (kind.HasFlag(BicepCompletionContextKind.Expression) &&
-                    PropertyTypeShouldFlowThrough(matchingNodes, propertyInfo, offset))
+                if (kind.HasFlag(BicepCompletionContextKind.Expression))
                 {
-                    kind |= BicepCompletionContextKind.PropertyValue;
+                    if (PropertyTypeShouldFlowThrough(matchingNodes, propertyInfo, offset))
+                    {
+                        kind |= BicepCompletionContextKind.PropertyValue;
+                    }
+
+                    var arrayItemInfo = SyntaxMatcher.FindLastNodeOfType<ArrayItemSyntax, ArrayItemSyntax>(matchingNodes);
+                    if (ArrayItemTypeShouldFlowThrough(matchingNodes, arrayItemInfo))
+                    {
+                        kind |= BicepCompletionContextKind.ArrayItem;
+                    }
                 }
             }
 
+            if (propertyInfo.node?.TryGetKeyText() == LanguageConstants.ResourceDependsOnPropertyName
```

Comment:
To simplify this top-level method, could we move this logic to a dedicated method? E.g. following the pattern above of:
```csharp
kind |= ConvertFlag(IsDependsOnContext(...), BicepCompletionContextKind.ResourceDependsOnSymbolicReference);
```